usr/lib/webapp-manager/common.py

```diff
@@ -164,9 +164,16 @@ def create_webapp(self, name, url, icon, category, browser, isolate_profile=True
                 # Firefox based
                 firefox_profiles_dir = FIREFOX_PROFILES_DIR if browser.browser_type == BROWSER_TYPE_FIREFOX else FIREFOX_FLATPAK_PROFILES_DIR
                 firefox_profile_path = os.path.join(firefox_profiles_dir, codename)
-                desktop_file.write("Exec=sh -c 'XAPP_FORCE_GTKWINDOW_ICON=" + icon + " " + browser.exec_path +
-                                    " --class WebApp-" + codename +
-                                    " --profile " + firefox_profile_path +
+                if privatewindow:
+                    desktop_file.write("Exec=sh -c 'XAPP_FORCE_GTKWINDOW_ICON=" + icon + " " + browser.exec_path +
+                                        " --private-window" +
+                                        " --class WebApp-" + codename +
+                                        " --profile " + firefox_profile_path +
+                                    " --no-remote " + url + "'\n")
+                else:
+                    desktop_file.write("Exec=sh -c 'XAPP_FORCE_GTKWINDOW_ICON=" + icon + " " + browser.exec_path +
+                                        " --class WebApp-" + codename +
+                                        " --profile " + firefox_profile_path +
                                     " --no-remote " + url + "'\n")
```

Comment:
FYI: What do you think about extract the exec string to a variable, and then just write it? Like:
```python
run_string = (
     "Exec=sh -c 'XAPP_FORCE_GTKWINDOW_ICON=" + icon + "
     " + browser.exec_path +
     " --class WebApp-" + codename +
     " --profile " + firefox_profile_path +
     " --no-remote "
)
if privatewindow:
    run_string += " --private-window "
desktop_file.write(run_string + url + "'\n")
```

Reply:
Thank you for the suggestion. To be frank, this is my first time writing a python code for gtk application. I would try that and create another pull request.